lib/watcher.js

```diff
@@ -173,10 +175,68 @@ Watcher.prototype.updateExclusivity = function (file, hasExclusiveTests) {
 	}
 };
 
+Watcher.prototype.trackFailures = function (api) {
```

Comment:
The current approach is relatively clean. We could do something like that if we're going to move more of the watcher functionality into the API maybe, or when we have a clearer idea of what programmatic interface we need to present to editors plugins.